Mixpanel/MPABTestDesignerConnection.m

```diff
@@ -218,21 +222,24 @@ - (void)webSocket:(MPWebSocket *)webSocket didCloseWithCode:(NSInteger)code reas
     _open = NO;
     if (_connected) {
         _connected = NO;
-        [self reconnect:YES];
+        [self reconnect:YES maxInterval:10 maxRetries:10];
         if (_disconnectCallback) {
             _disconnectCallback();
         }
+    } else if (_keepTrying) {
+        [self reconnect:YES maxInterval:30 maxRetries:40];
     }
 }
 
-- (void)reconnect:(BOOL)initiate
+- (void)reconnect:(BOOL)initiate maxInterval:(int)maxInterval maxRetries:(int)maxRetries
 {
     static int retries = 0;
-    if (self.sessionEnded || _connected || retries >= 10) {
+    
+    if (self.sessionEnded || _connected || retries >= maxRetries) {
         // If we deliberately closed the connection, or are already connected
         // or we tried too many times, then stop retrying.
         retries = 0;
-    } else if(initiate ^ (retries > 0)) {
+    } else if (initiate ^ (retries > 0)) {
```

Comment:
This is just me being a tourist, but what's the deal with the caret operator here? I want to read this as `bitwise xor initiate and (retries > 0)` - is it the case that YES and NO are bitwise compliments, so we only run the block if exactly one of these is YES and the other is NO? Or does the binary caret operator mean other stuff in Objective C?